biolearn/data_library.py

```diff
@@ -405,6 +415,162 @@ def _metadata_load_list(self):
+        for key in keys:
+            if key not in data:
+                raise ValueError(f"Parser not valid: missing {key}")
+
+    def _create_metadata(self, metadata_query_url):
+        response = requests.get(metadata_query_url)
+        json_data = response.json()
+        smaples = [item for item in json_data['GeoMetaData']]
+        return self._convert_to_metadata_df(self.metadata_keys, smaples)
+
+    def _map_characteristics_to_dict(self, sample):
+
+        def extract_key(item):
+            if 'tag' in item:
+                return item['tag']
+            else:
+                return item['value'].split(":")[0]
+
+        def extract_value(item):
+            if 'tag' in item:
+                return item['value']
+            else:
+                return item['value'].split(":")[1]
+
+        characteristics = sample["entity"]["sample"]["channels"][0]["characteristics"]
+
+        return {extract_key(char).lower(): extract_value(char) for char in characteristics}
+
+    def _convert_characteristics_to_df_cols_data(self, metadata_keys, item):
+        cols_data = []
+
+        characteristics = self._map_characteristics_to_dict(item)
+
+        for key in metadata_keys:
+            parse_type = self.metadata_keys_parse[key]
+            if parse_type == "sex":
+                cols_data.append(sex_parser(characteristics.get(key)))
+            elif parse_type == "numeric":
+
+                if key == "age":
+                    if key in characteristics:
+                        cols_data.append(extract_numeric(characteristics[key]))
+                    else:
+                        cols_data.append(extract_informal_age(characteristics))
+                else:
+                    cols_data.append(extract_numeric(characteristics.get(key, "")))
+            else:
+                cols_data.append(characteristics.get(key))
+
+        return cols_data
+
+    def _convert_to_metadata_df(self, metadata_keys, samples):
+
+        data = {sample["acc"]: self._convert_characteristics_to_df_cols_data(
+            metadata_keys, sample) for sample in samples}
+
+        df = pd.DataFrame(data).transpose()
+        df = df.reset_index(drop=False)
+        columns = ["id"]
+        columns.extend(metadata_keys)
+        df.columns = columns
+        return df
+
+    def _get_matrix_file_size_from_url(self, url):
+        try:
+            response = requests.head(url)
+            response.raise_for_status()
+
+            content_length = response.headers.get('Content-Length')
+
+            if content_length is None:
+                print("Content-Length header is not present.")
+                return None
+
+            file_size = int(content_length)
+            return file_size
+
+        except requests.RequestException as e:
+            print(f"HTTP request failed: {e}")
+            return None
+
+    def _create_matrix(self, matrix_file):
+        matrix_file_path = cached_download(matrix_file)
```

Comment:
This is my fault since I didn't document this but cached_download is deprecated in favor of the classes in cache.py that allow for configuration. I don't think it needs to block merge but we will need to resolve before merging the library files. Support for what you need here would require some feature additions to the cache class to support temporary file caching. No need to make any changes just a note for the future.